ratatui-widgets/src/calendar.rs

```diff
@@ -302,4 +302,15 @@ mod tests {
     fn test_today() {
         CalendarEventStore::today(Style::default());
     }
+
+    #[test]
+    fn buffer_overflow() {
+        let mut buffer = Buffer::empty(Rect::new(0, 0, 1, 1));
+        let calendar = Monthly::new(
+            Date::from_calendar_date(2023, Month::January, 1).unwrap(),
```

Comment:
can we avoid unwrap here? Maybe expect it or do proper error handling.

Also:

```suggestion
            Date::from_calendar_date(1984, Month::January, 1).unwrap(),
```

Reply:
Unwrap is often fine in tests. It generally indicates that you expect that call never to fail, and if it does then the test is wrong and you mainly should be looking at a stack trace etc.. Not sure what the linter does with it from memory.